src/main/java/com/github/kagkarlsson/scheduler/task/Task.java

```diff
@@ -53,5 +63,31 @@ public String toString() {
 		return "Task " +
 				"task=" + getName();
 	}
+
+	public interface Serializer<T> {
+		byte[] serialize(T data);
+		T deserialize(byte[] serializedData);
+
+		Serializer JAVA_SERIALIZER = new Serializer<Object>() {
+			public byte[] serialize(Object data) {
```

Comment:
Should this have been made generic as well? Trying to figure out where the cast is otherwise.. :)